Core/source/core/pgp-key.ts

```diff
@@ -172,21 +174,23 @@ export class PgpKey {
       throw new Error(`No private key packets in key to encrypt. Is this a private key?`);
     }
     if (encryptedPacketCount) {
-      throw new Error(`Cannot encrypt a key that has ${encryptedPacketCount} of ${secretPackets.length} private packets still encrypted`);
+      throw new Error(`Cannot encrypt a key that has ${encryptedPacketCount} of ` +
+        `${secretPackets.length} private packets still encrypted`);
     }
-    await prv.encrypt(passphrase);
+    await ((prv as unknown) as BaseSecretKeyPacket).encrypt(passphrase);
```

Comment:
This looks wrong - `prv` here should be of a `Key` type. `BaseSecretKeyPacket` is not of `Key` type, `PrivateKey` would be appropriate.

Reply:
fixed